falco/templates/pod-template.tpl

```diff
@@ -0,0 +1,326 @@
+    {{- if and .Values.certs (not .Values.certs.existingSecret) }}
+    checksum/certs: {{ include (print $.Template.BasePath "/certs-secret.yaml") . | sha256sum }}
+    {{- end }}
+    {{- with .Values.podAnnotations }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  serviceAccountName: {{ include "falco.serviceAccountName" . }}
+  {{- with .Values.podSecurityContext }}
+  securityContext:
+    {{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- if .Values.driver.enabled }}
+  {{- if and (eq .Values.driver.kind "ebpf") .Values.driver.ebpf.hostNetwork }}
+  hostNetwork: true
+  dnsPolicy: ClusterFirstWithHostNet
+  {{- end }}
+  {{- end }}
+  {{- if .Values.priorityClassName }}
+  priorityClassName: {{ .Values.priorityClassName }}
+  {{- end }}
+  {{- with .Values.nodeSelector }}
+  nodeSelector:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.affinity }}
+  affinity:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.tolerations }}
+  tolerations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.imagePullSecrets }}
+  imagePullSecrets: 
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  containers:
+    - name: {{ .Chart.Name }}
+      image: {{ include "falco.image" . }}
+      imagePullPolicy: {{ .Values.image.pullPolicy }}
+      resources:
+        {{- toYaml .Values.resources | nindent 8 }}
+      securityContext:
+        {{- include "falco.securityContext" . | nindent 8 }}
+      args:
+        - /usr/bin/falco
+        {{- with .Values.collectors }}
+        {{- if .enabled }}
+        {{- if .containerd.enabled }}
+        - --cri
+        - /run/containerd/containerd.sock
+        {{- end }}
+        {{- if .crio.enabled }}
+        - --cri
+        - /run/crio/crio.sock
+        {{- end }}
+        {{- if .kubernetes.enabled }}
+        - -K
+        - {{ .kubernetes.apiAuth }}
+        - -k
+        - {{ .kubernetes.apiUrl }}
+        {{- if .kubernetes.enableNodeFilter }}
+        - --k8s-node
+        - "$(FALCO_K8S_NODE_NAME)"
+        {{- end }}
+        {{- end }}
+        - -pk
+        {{- end }}
+        {{- end }}
+    {{- with .Values.extra.args }}
+      {{- toYaml . | nindent 8 }}
+    {{- end }}
+      env:
+        - name: FALCO_K8S_NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+      {{- if or (not .Values.driver.enabled) (and .Values.driver.loader.enabled .Values.driver.loader.initContainer.enabled) }}
+        - name: SKIP_DRIVER_LOADER
+          value:
+      {{- end }}
+      {{- if and .Values.driver.enabled (eq .Values.driver.kind "ebpf") }}
+        - name: FALCO_BPF_PROBE
+          value: {{ .Values.driver.ebpf.path }}
+      {{- end }}
+      {{- range $key, $value := .Values.extra.env }}
+        - name: "{{ $key }}"
+          value: "{{ $value }}"
+      {{- end }}
+      {{- if .Values.falco.webserver.enabled }}
+      livenessProbe:
```

Comment:
Do we really need to expose the values for `livenesProbe` and `readynessProbe` in the `values.yaml `? Since the embedded webserver is used only for this purpose I don't see a scenario where it could be under heavy load and miss the `probe requests`. Default values should work fine on most scenarios, and if `Falco pods` get restarted due to `liveness probes` not getting answered then  something is wrong and needs to be investigated by the user. It's just a thought, let me know what you think.

Reply:
I partially agree with you, but I'd leave them since they can be helpful in extreme situation (for example, a very slow CI).